virtcontainers/clh.go

```diff
@@ -861,439 +784,165 @@ func (clh *cloudHypervisor) LaunchClh() (string, int, error) {
 
-	cmdline.args = append(cmdline.args, cscKernel)
-	cmdline.args = append(cmdline.args, o.path)
-
-}
+	cfg := chclient.NewConfiguration()
 
-//****************************************
-// The Path to the root (boot) disk image
-//****************************************
-type CLIDisk struct {
-	path  string
-	iommu bool
-}
+	socketTransport := &http.Transport{
+		DialContext: func(ctx context.Context, network, path string) (net.Conn, error) {
+			addr, err := net.ResolveUnixAddr("unix", c.state.apiSocket)
+			if err != nil {
+				return nil, err
 
-func (o *CLIDisk) Build(cmdline *CommandLine) {
+			}
 
-	cmdline.args = append(cmdline.args, cscDisk)
-	if o.iommu {
-		cmdline.args = append(cmdline.args, "path="+o.path+",iommu=on")
-	} else {
-		cmdline.args = append(cmdline.args, "path="+o.path+",iommu=off")
+			return net.DialUnix("unix", nil, addr)
+		},
 	}
 
-}
+	cfg.HTTPClient = http.DefaultClient
+	cfg.HTTPClient.Transport = socketTransport
 
-//****************************************
-// The random device
-//****************************************
-type CLIRng struct {
-	src   string
-	iommu bool
+	return chclient.NewAPIClient(cfg).DefaultApi
 }
 
-func (o *CLIRng) Build(cmdline *CommandLine) {
+func openApiClientError(err error) error {
 
-	cmdline.args = append(cmdline.args, cscRng)
-	if o.iommu {
-		cmdline.args = append(cmdline.args, "src="+o.src+",iommu=on")
-	} else {
-		cmdline.args = append(cmdline.args, "src="+o.src+",iommu=off")
+	if err == nil {
+		return nil
 	}
 
-}
-
-//****************************************
-// The VSock socket
-//****************************************
-type CLIVsock struct {
-	socketPath string
-	cid        uint32
-	iommu      bool
-}
-
-func (o *CLIVsock) Build(cmdline *CommandLine) {
-
-	cmdline.args = append(cmdline.args, cscVsock)
-	if o.iommu {
-		cmdline.args = append(cmdline.args, "cid="+strconv.FormatUint(uint64(o.cid), 10)+",sock="+o.socketPath+",iommu=on")
-	} else {
-		cmdline.args = append(cmdline.args, "cid="+strconv.FormatUint(uint64(o.cid), 10)+",sock="+o.socketPath+",iommu=off")
+	reason := ""
+	if apierr, ok := err.(chclient.GenericOpenAPIError); ok {
+		reason = string(apierr.Body())
 	}
-}
 
-//****************************************
-// The shard (virtio) file system
-//****************************************
-type CLIFs struct {
-	tag        string
-	socketPath string
-	queues     uint32
-	queueSize  uint32
-	dax        bool
+	return fmt.Errorf("error: %v reason: %s", err, reason)
 }
 
-func (o *CLIFs) Build(cmdline *CommandLine) {
-
-	cmdline.args = append(cmdline.args, cscFs)
-
-	fsarg := "tag=" + o.tag + ",sock=" + o.socketPath
-	if o.dax {
-		fsarg += ",dax=on"
-	} else {
-		fsarg += ",num_queues=" + strconv.FormatUint(uint64(o.queues), 10) + ",queue_size=" + strconv.FormatUint(uint64(o.queueSize), 10)
+func (c *cloudHypervisor) bootVM(timeout int) error {
+	if timeout < 0 {
+		return fmt.Errorf("Invalid timeout %ds", timeout)
```

Comment:
Great that we are checking this, but the code seems to be littered with this check. Could you find a way to wrap it in a function or just perform the check once so that other functions can rely on the timeout?